pkg/deploy/request.go

```diff
@@ -111,6 +114,10 @@ func getFiles(dirFileNames []string) ([]string, error) {
 	return files, nil
 }
 
+func isHTTPUrl(fileName string) bool {
```

Comment:
Deploy Engine uses `govalidator` for a few things like this. Not sure if it's worth it but they handle a lot of edge cases I wouldn't have thought of: https://github.com/asaskevich/govalidator/blob/master/validator.go#L104-L125

Also `IsFilePath`: https://github.com/asaskevich/govalidator/blob/master/validator.go#L585-L597

Reply:
We caught up offline, but following up here. This validator is too lax on URL checking because it accepts relative paths for URLs. It also does not see FilePaths which are relative as valid.